### PR TITLE
Make kernel options more robust

### PR DESCRIPTION
**Pull request by isantop**
Users or external programs may modify the kernelstub configuration file directly, instead of relying on the built-in commands that kernelstub provides. The most common modification affects the `kernel_options` setting and saves this as a string instead of a list. This should be avoided, but is largely recoverable by running this string through the built-in option parser and re-saving the configuration.

This PR syncs with upstream, and includes a fix for this issue. Upon detecting a string-type `kernel_options` setting in an current config_rev, kernelstub will attempt to fix the issue by running said string through the options parser and resetting `kernel_options` to a list. It will also print a warning informing the user about the problem with possible remedies.

If the fix fails, kernelstub will raise an exception, print the above error message, then exit with status 169 (Malformed configuration).

### CODE REVIEW

File: kernelstub/config.py [hunk 1]
```diff
@@ -106,8 +126,10 @@ def update_config(self, config):
             config['user']['live_mode'] = False
             config['default']['live_mode'] = False
         if config['user']['config_rev'] < 3:
-            config['user']['kernel_options'] = self.parse_options(config['user']['kernel_options'].split())
-            config['default']['kernel_options'] = self.parse_options(config['default']['kernel_options'].split())
+            if type(config['user']['kernel_options']) is str:
+                config['user']['kernel_options'] = self.parse_options(config['user']['kernel_options'].split())
+            if type(config['default']['kernel_options']:
```
Comment (jackpot51) [inline kernelstub/config.py]: Missing a parens. Please check syntax with pyflakes3

File: kernelstub/config.py [hunk 2]
```diff
@@ -84,6 +84,26 @@ def load_config(self):
                 self.log.info("Configuration updated successfully!")
             elif self.config['user']['config_rev'] == self.config_default['default']['config_rev']:
                 self.log.debug("Configuration up to date")
+                #Double-checking in case OEMs do bad things with the config file
+                if type(config['user']['kernel_options']) is str:
+                    try:
+                        config['user']['kernel_options'] = self.parse_options(config['user']['kernel_options'].split())
+                        self.log.warning('Malformed configuration file found!\n\n'
+                                         'The configuration file has the incorrect format for the kernel options. '
```
Comment (jackpot51) [inline kernelstub/config.py]: This is a very long error message. Consider shortening it some.

File: kernelstub/config.py [hunk 3]
```diff
@@ -84,6 +84,26 @@ def load_config(self):
...
+                                         'Usually this is caused by manually editing the configuration file, but '
+                                         'it can also be caused by outdated or buggy maintainer packages from your '
+                                         'hardware OEM. If you haven\'nt manually edited the config file, then '
+                                         'contact your hardware vendor to inform them to fix their packages.\n\n'
+                                         'The issue was able to be corrected automatically, and the corrected '
+                                         'configuration will be saved')
+                    except:
+                        raise ConfigError('Malformed configuration file found!\n\n'
+                                          'The configuration file has the incorrect format for the kernel options. '
```
Comment (jackpot51) [inline kernelstub/config.py]: Don't repeat yourself.

**Comment by brs17** [comment]
It would be wonderful if you could merge the upstream code into a branch of pop-os/kernelstub and make the pr from that branch->master. That way the CI can run. Thank you.

**Comment by brs17** [comment]
Well I made one: https://github.com/pop-os/kernelstub/pull/27

**Comment by jackpot51** [comment]
Closing in favor of #27

**Pull request closed without merging**